src/Model/Join.php

```diff
@@ -132,6 +131,44 @@ public function __construct(string $foreign_table = null)
         }
     }
 
+    /**
+     * Create fake foreign model, in the future, this method should be removed
+     * in favor of always requiring an object model.
+     */
+    protected function createFakeForeignModel(): Model
```

Comment:
FYI if foreign model hooks needs to be executed, override this method to provide non-faked Model for `$this->foreign_table` table